### PR TITLE
User secure endpoints

### PR DESCRIPTION
**Pull request by aag7593**
add, update and delete users as an admin

### CODE REVIEW

File: routes/user/userSecure.js [hunk 1]
```diff
@@ -90,4 +92,135 @@ router.put('/permission', async (req, res) => {
...
 });
 
+// Create new user
+router.post('/', async (req, res) => {
+    try {
+        if (req.body && req.body.firstName && req.body.lastName && req.body.roleID && req.body.email && req.body.password) {
+            if (req.user.roleID == 14139 || req.user.roleID == 21149) {
+                const { firstName, lastName, roleID, email, password } = req.body;
+
+                if (!validateName(firstName)) {
```
Comment (vsoler27) [inline routes/user/userSecure.js:102]: a lot of these error codes are wrong a 400 means bad request/bad data and a 403 means forbidden/incorrect permissions

File: routes/user/userSecure.js [hunk 2]
```diff
@@ -90,4 +92,135 @@ router.put('/permission', async (req, res) => {
...
+                        res.status(400).json({'error': 'Email Provided does not have a university domain registered in the system! Contact your university for more information.'});
+                    }
+                }
+            }
+            else {
+                res.status(401).json({'error': "you are not authorized to complete this action"});
+            }
+        }
+        else {
+            console.log("CRAP");
```
Comment (vsoler27) [inline routes/user/userSecure.js]: pls remove

File: routes/user/userSecure.js [hunk 3]
```diff
@@ -90,4 +92,135 @@ router.put('/permission', async (req, res) => {
...
+                            res.status(200).json({updated});
+                        }
+                    });
+                }
+                else {
+                    res.status(404).json({"error": "User Not Found"});
+                }
+            }
+            else {
+                res.status(404).json({"error": "Incomplete Input"});
```
Comment (vsoler27) [inline routes/user/userSecure.js]: make this one a 400

**Review by vsoler27 — CHANGES_REQUESTED** [review]
requesting some small changes but overall this looks good

File: routes/user/userSecure.js (hunk 1 above)
Comment (aag7593) [reply routes/user/userSecure.js:102]: Ah okay- I'll make those changes

File: routes/user/userSecure.js (hunk 2 above)
Comment (aag7593) [reply routes/user/userSecure.js]: oop

**Review by vsoler27 — APPROVED** [review]

**Pull request merged**